### ISSUE TITLE
Suggestion

### ISSUE THREAD

**Issue by GreatWall7538** [sp614x/optifine#7130]
Please open source OptiFine!
It's OptiWorst for some Forge mods & Fabric mods because extremely poor compatibility!

**Comment by Ari-RERA-43** [comment]
Here's the FAQ entry for this question (in the Discord): 

>>> Q: **Why isn't OptiFine open-source?**
A: The core of OptiFine consists of many, various changes to Minecraft's rendering code. Rather than simple patches, these are significant reorganizations. This means publishing the full source code of OptiFine would be a direct violation of Minecraft's EULA. Technically, it would be possible to extract the actual changes as patches, which can then be published to GitHub as source code. However, there are a number of issues with this idea. First, OptiFine is built on a custom version of Mod Coder Pack. This non-standard version of MCP is used to allow OptiFine to start development on new versions of the game much, much earlier. During this time, official MCP mappings are either completely missing, or otherwise very unstable. Unfortunately, according to the MCP license and Terms of Usage, modified versions of MCP scripts are not allowed to be distributed. This means that, even if OptiFine patches were released, nobody else would be able to collaborate on the code, defeating the entire purpose of going open-source. Even if somehow *all of that* was solved, this would still mean significantly changing how OptiFine development is handled. sp614x does not work on the mod using patches, which means he would have to either merge the patches manually, or completely change his entire workflow to use patches. Changing the development process like this would not be a trivial task in the slightest.

In summary, due to a *multitude* of legal and technical complications, sp614x cannot make OptiFine open-source.

**Comment by ZenIsBestWolf** [comment]
see above thanks aria
